### ISSUE TITLE
Clean up direct access to MOFWBEMConnection instance variables

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#369]
The code in `test_mof_compiler.py` accesses instance variables of `MOFWBEMConnection` directly.
They were not meant to be part of the public interface, I think. Also, even if we make them part of the public interface, their names are not currently very consistent and should be fixed, plus they should become properties that are read-only.

Let's discuss.

This will be cleaned up as part of issue #2062. Update: It has not.

**Comment by andy-maier** [comment]
This issue needs to be re-evaluated once the mocker refactoring is complete.

**Comment by andy-maier** [comment]
There are presently two occurrences of direct instance attribute access to `MOFWBEMConnection` in test_mof_compiler.py, in `test_remove()`:

```
        conn_mof = MOFWBEMConnection(conn=conn)
        conn_mof.default_namespace = conn.default_namespace
        conn_mof.url = conn.url
```

Analysis:
* `default_namespace` is a settable property. However, in this case it does not need to be set because it defaults to the value it is set to. So the setting of it can be removed.
* `url` does not exist on `MOFWBEMConnection` and is not used in the rest of this test function. So the setting of it can be removed.

**Comment by andy-maier** [comment]
Addressed in PR #2239.
